### ISSUE TITLE
number-unit-format reads a hyphen as a unit, so "2021-06" and "5-10 mm" are unsatisfiable false positives

### ISSUE THREAD

**Issue by Jamie-BitFlight** [Jamie-BitFlight/textlint-rule-preset-ste-ai#148]
## Symptom

`ste-ai/number-unit-format` treats the text after a hyphen as a unit. A hyphenated date, and a
numeric range, both produce a finding whose suggested repair is not a number.

```
The release shipped in 2021-06 without issue.
  → Put a space between the number and the unit: "2021 -06".

The range is 5-10 mm wide here.
  → Put a space between the number and the unit: "5 -10 mm".
```

Neither repair can be applied. `2021 -06` is not a date, and `5 -10 mm` is not a range.

## Relationship to #141

This is the same family as #141, reached through a different character. #141 was the decimal
separator: `3.11` read as `3` + `.11`. This is the hyphen.

PR #147 fixes #141 by forbidding a unit to start with `.` or `,`. It deliberately does not touch
the hyphen, because the hyphen needs its own decision that #141 did not raise. That PR's change
leaves this defect exactly as it was.

## Where

`src/deterministic/rules/mechanics.ts`, `numberUnitFormatRule`. The unit capture is
`([^\s\d.,].*)` after PR #147, and `([^\s\d].*)` before it. A hyphen satisfies both.

The regions reach the rule from `src/core/protected-regions.ts`'s `numeric-expression` pass. Its
range pattern captures `5-10`, and its version pattern captures a trailing `-<identifier>` segment.

## Reproduction

```bash
npx -p vite-plus@0.3.0 vp install --frozen-lockfile
node_modules/.bin/vp pack
printf 'The release shipped in 2021-06 without issue.\n\nThe range is 5-10 mm wide here.\n' > probe.md
node dist/cli/main.js lint probe.md --config .ste-ai.json --deterministic-only
```

Observed on the head of PR #147, which already carries the #141 fix. Both findings are
`deterministic-violation`.

## How it was found

While fixing #141. I enumerated every corpus annotation that expects `number-unit-format`, to find
which expectations the #141 fix invalidates. Nine of the ten stopped firing. The tenth,
`sqlite-cli-dot-commands-p5` on the quote `2021-06`, still fires. That annotation is recorded as
`disputed`, so a reviewer had already judged it a false positive. PR #147 leaves that record intact
precisely because the finding is still real.

## The design question this needs

A hyphen is not like a decimal separator. It carries at least three meanings next to a number:

1. A range: `5-10 mm`. The quantity is the whole range, and the real unit follows it.
2. A date or version segment: `2021-06`, `1.2.3-beta`. There is no unit at all.
3. A signed value: `-10`, where the hyphen belongs to the number rather than after it.

Excluding `-` from the unit's first character, the way PR #147 excludes `.` and `,`, would silence
cases 1 and 2 together. It would also stop the rule ever checking spacing on the unit that follows
a range, which case 1 does have. Whether that trade is right is a decision for the rule's owner,
which is why this is not folded into PR #147.

## Suggested acceptance criteria

- `The release shipped in 2021-06 without issue.` produces no finding.
- `The range is 5-10 mm wide here.` produces no finding that suggests `5 -10 mm`.
- `Torque the bolt to 25Nm now.` still flags the missing space.
- `The file is 5.6 MB in size.` still produces no finding.
- The decision about whether a range's own unit spacing stays checked is recorded in
  `docs/provisional-rules.md`, with a test pinning whichever behaviour is chosen.
